### PR TITLE
Drop Python 3.8 support

### PR DESCRIPTION
**Pull request by mudit2812**
As title says

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,28 @@
+# Release 0.32.0-dev
+
+### New features since last release
+
+### Improvements 🛠
+
+### Breaking changes 💔
+
+* Support for Python 3.8 has been removed.
+  [(#)]()
```
Comment (mudit2812) [inline CHANGELOG.md]:
```suggestion
  [(#40)](https://github.com/PennyLaneAI/pennylane-orquestra/pull/40)
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -17,7 +17,6 @@ jobs:
     strategy:
       matrix:
         config:
-          - {python-version: 3.8}
           - {python-version: 3.9}
           - {python-version: '3.10'}
```
Comment (timmysilv) [inline .github/workflows/tests.yml:21]:
```suggestion
          - {python-version: '3.10'}
          - {python-version: '3.11'}
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,28 @@
+# Release 0.32.0-dev
+
+### New features since last release
+
+### Improvements 🛠
+
+### Breaking changes 💔
+
+* Support for Python 3.8 has been removed.
```
Comment (timmysilv) [inline CHANGELOG.md]:
```suggestion
* Support for Python 3.8 has been removed, and support for Python 3.11 has been added.
```

**Review by timmysilv — COMMENTED** [review]
let's run it with 3.11 and see if we can add support there. seems to be the trend in other plugins

**Review by timmysilv — APPROVED** [review]

**Pull request merged**